src/coreclr/jit/emitxarch.cpp

```diff
@@ -749,6 +749,9 @@ bool emitter::TakesRexWPrefix(instruction ins, emitAttr attr)
             case INS_pdep:
             case INS_pext:
             case INS_rorx:
+            case INS_shlx:
```

Comment:
As per the changes at other places, we have the code to have it supported only for `TARGET_64BIT`, but here, it is under `TARGET_ARM64`. What is the difference and should it be consistent?

Reply:
It is under #ifdef TARGET_AMD64, not ARM64. So, I guess it is correct.